pkg/controller/podprobemarker/podprobemarker_event_handler.go

```diff
@@ -91,51 +92,86 @@ func (p *enqueueRequestForPod) Update(ctx context.Context, evt event.UpdateEvent
 	if !ok {
 		return
 	}
-	// add pod probe to nodePodProbe.spec
 	oldInitialCondition := util.GetCondition(old, corev1.PodInitialized)
 	newInitialCondition := util.GetCondition(new, corev1.PodInitialized)
-	if newInitialCondition == nil {
+	if !kubecontroller.IsPodActive(new) || newInitialCondition == nil ||
+		newInitialCondition.Status != corev1.ConditionTrue || new.Spec.NodeName == "" {
 		return
 	}
-	if !kubecontroller.IsPodActive(new) {
+
+	isServerlessPod := false
+	node := &corev1.Node{}
+	if err := p.reader.Get(context.TODO(), client.ObjectKey{Name: new.Spec.NodeName}, node); err != nil {
+		klog.ErrorS(err, "Failed to get Node", "nodeName", new.Spec.NodeName)
 		return
 	}
-	if ((oldInitialCondition == nil || oldInitialCondition.Status == corev1.ConditionFalse) &&
-		newInitialCondition.Status == corev1.ConditionTrue) || old.Status.PodIP != new.Status.PodIP {
-		ppms, err := p.getPodProbeMarkerForPod(new)
-		if err != nil {
-			klog.ErrorS(err, "Failed to List PodProbeMarker")
-			return
+	if node.Labels["type"] == VirtualKubelet {
+		isServerlessPod = true
+	}
+
+	triggerReconcile := false
+	diff := sets.NewString()
+	// normal pod
+	if !isServerlessPod {
+		if ((oldInitialCondition == nil || oldInitialCondition.Status == corev1.ConditionFalse) &&
+			newInitialCondition.Status == corev1.ConditionTrue) || old.Status.PodIP != new.Status.PodIP {
+			triggerReconcile = true
+		}
+	} else if utilfeature.DefaultFeatureGate.Enabled(features.EnablePodProbeMarkerOnServerless) {
+		diff = diffPodConditions(old.Status.Conditions, new.Status.Conditions)
```

Comment:
add a comment about why reconcile is required for serverless pod once condition is changed